### PR TITLE
Zustand typescript examples

### PR DESCRIPTION
**Pull request by GuillaumeSalles**
Migrate zustand examples to typescript to help the community with the generic types.

- [x] Migrate `zustand-todo-list` example to typescript
- [x] Migrate `zustand-whiteboard` example to typescript

### CODE REVIEW

File: examples/zustand-todo-list/package.json [hunk 1]
```diff
@@ -2,19 +2,33 @@
...
-    "@liveblocks/zustand": "^0.17.7",
-    "react": "^18.1.0",
-    "react-dom": "^18.1.0",
-    "react-scripts": "^5.0.1",
-    "zustand": "^3.7.1"
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
```
Comment (marcbouchenoire) [inline examples/zustand-todo-list/package.json:31]: We don't use ESLint in any examples yet and when we do it'll be better to do it in one go and figure out a shared config, for now we should remove any specific configs.

File: examples/zustand-whiteboard/package.json [hunk 2]
```diff
@@ -19,6 +24,12 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
```
Comment (marcbouchenoire) [inline examples/zustand-whiteboard/package.json:32]: Same as above.

**Review by marcbouchenoire — CHANGES_REQUESTED** [review]
Nice, looks good to me! I just added comments about removing ESLint configs until we figure this out across examples.

**Comment by GuillaumeSalles** [comment]
> Nice, looks good to me! I just added comments about removing ESLint configs until we figure this out across examples.

Nice catch :) done

**Review by marcbouchenoire — APPROVED** [review]

**Pull request merged**